nats/aio/transport.py

```diff
@@ -140,44 +137,25 @@ async def connect_tls(
         ssl_context: ssl.SSLContext,
         buffer_size: int,
         connect_timeout: int,
-    ):
-        # loop.start_tls was introduced in python 3.7
-        # the previous method is removed in 3.9
-        if sys.version_info.minor >= 7:
-            # manually recreate the stream reader/writer with a tls upgraded transport
-            reader = asyncio.StreamReader()
-            protocol = asyncio.StreamReaderProtocol(reader)
-            transport_future = asyncio.get_running_loop().start_tls(
-                self._io_writer.transport,
-                protocol,
-                ssl_context,
-                # hostname here will be passed directly as string
-                server_hostname=uri if isinstance(uri, str) else uri.hostname
-            )
-            transport = await asyncio.wait_for(
-                transport_future, connect_timeout
-            )
-            writer = asyncio.StreamWriter(
-                transport, protocol, reader, asyncio.get_running_loop()
-            )
-            self._io_reader, self._io_writer = reader, writer
-        else:
-            transport = self._io_writer.transport
-            sock = transport.get_extra_info('socket')
-            if not sock:
-                # This shouldn't happen
-                raise errors.Error('nats: unable to get socket')
-
-            connection_future = asyncio.open_connection(
-                limit=buffer_size,
-                sock=sock,
-                ssl=ssl_context,
-                # hostname here will be passed directly as string
-                server_hostname=uri if isinstance(uri, str) else uri.hostname,
-            )
-            self._io_reader, self._io_writer = await asyncio.wait_for(
-                connection_future, connect_timeout
-            )
+    ) -> None:
+        assert self._io_writer, f'{type(self).__name__}.connect must be called first'
+
+        # manually recreate the stream reader/writer with a tls upgraded transport
+        reader = asyncio.StreamReader()
+        protocol = asyncio.StreamReaderProtocol(reader)
+        transport_future = asyncio.get_running_loop().start_tls(
+            self._io_writer.transport,
+            protocol,
+            ssl_context,
+            # hostname here will be passed directly as string
+            server_hostname=uri if isinstance(uri, str) else uri.hostname
+        )
+        transport = await asyncio.wait_for(transport_future, connect_timeout)
+        assert isinstance(transport, asyncio.WriteTransport)
```

Comment:
This fails because apparently for uvloop, `uvloop.loop._SSLProtocolTransport` is not an instance of `asyncio.WriteTransport`. I will remove these asserts before merging for now, tests passing locally otherwise.

Reply:
oh I see but then with mypy start to get the following...
```
nats/aio/transport.py:170: error: Item "None" of "Optional[StreamReader]" has no attribute "read"  [union-attr]
```